docs/concepts/crds/backupsession.md

```diff
@@ -0,0 +1,139 @@
+    - directory: /source/data
+      fileStats:
+        modifiedFiles: 0
+        newFiles: 0
+        totalFiles: 2
+        unmodifiedFiles: 2
+      name: ea5381cc
+      processingTime: 0m0s
+      size: 30 B
+      uploaded: 0 B
+```
+
+Here, we are going to describe some important sections of `BackupSession` crd.
+
+### BackupSession `Metadata`
+
+#### metadata.name
+
+`metadata.name` indicates the name of the `BackupSession`. This name is automatically generated by respective `CronJob` and it follows the following pattern: `<BackupConfiguration name>-<random suffix>`.
+
+#### metadata.namespace
+
+`metadata.namespace` indicates the name of the `BackupSession`. It is same as the namespace of respective `BackupConfiguration` object.
+
+#### metadata.labels
+
+`metadata.labels` holds respective `BackupConfiguration` name as label. Stash backup sidecar container use this label to watch only the BackupSessions of that `BackupConfiguration`.
+
+>If you create `BackupSession` manually to trigger a backup instantly, make sure that you have added `backup-configuration: <BackupConfiguration name>` label to your `BackupSession`. Otherwise, it will not trigger backup for workloads (those resources that are backed up using sidecar).
+
+### BackupSession `Spec`
+
+A `BackupSession` object has the following fields in `spec` section:
+
+#### spec.backupConfiguration
+
+`spec.backupConfiguration.name` indicates the name of the `BackupConfiguration` object whose target will be triggered to take backup instantly by this `BackupSession`.
+
+### BackupSession `Status`
+
+`.status` section of `BackupSession` shows progress, stats and overall phase of backup process in this session. Backup sidecar container or job update their respective filed of this `.status` after they complete their task. `.status` section consist of following fields:
+
+#### status.phase
+
+`status.phase` indicates the overall phase of the backup process for this BackupSession. `status.phase` will be `Succeeded` only if the phase of all hosts are `Succeeded`. If any of the hosts fail to complete backup, `status.phase` will be `Failed`.
+
+#### status.totalHosts
+
+A `BackupSession` may trigger backup of multiple hosts. For example, all the pod's of a `Deployment`, `ReplicaSet` and `ReplicationController` mounts same volume. In this case, Stash will backup data only from one pod. Thus, the total number of hosts for these workloads will be 1. On the other hand, pods of `StatefulSet` and `DaemonSet` may have different volume mounted into different replica. In this case, Stash will backup data from all individual pods. Thus, the total number of hosts for these workloads will be number of replicas for `StatefulSet` and number of running daemon pods for `DaemonSet`.
+
+#### status.sessionDuration
+
+`status.sessionDuration` indicates the total time taken to complete backup of all hosts in this session. It is simply the sum of backup duration of all individual hosts.
+
+#### status.stats
```

Comment:
I think this needs to clarify that this will be only available for driver = Restic. What about the other driver?